### DISCUSSION TITLE
Generate Token Endpoint /api/v1/account/<username>/token

### DISCUSSION THREAD

**Discussion by sass1997** [argoproj/argo-cd#12473] [Q&A]
I want to create a JWT Token for my My User with the Capabilities of apiKey (Documentation: apiKey - allows generating API keys).
This creation of the JWT Token I want to do with the username and password of my user. Since all other Logins are disabled due to SSO enforcing for the human users.

I also tried the same with argocd cli but this is also not working.

**Comment by jannfis** [comment]
The `apiKey` capability does not allow the generation of tokens, but the usage of the API using a token instead of user/pass authentication.

In order to generate an API key, a user needs to have RBAC permissions for the `account` resource, so e.g.

```
p, <somerole>, accounts, update, <someuser>, allow
```

**Reply by sass1997** [reply]
Do you have any idea then to automate this without any human clicks on the UI. We need to have such a token directly after argocd is installed since it's needed to bootstrap all the rest of the components.

**Reply by jannfis** [reply]
Please see https://github.com/argoproj/argo-cd/discussions/12486#discussioncomment-4997162 - there was a similar discussion going on in parallel :)
